### ISSUE TITLE
Add Dropout to PointwiseMLP

### ISSUE THREAD

**Issue by paulgekeler** [paulgekeler/neojax#11]
`neuraloperator` supports dropout in the `ChannelMLP`, see [here](https://github.com/neuraloperator/neuraloperator/blob/main/neuralop/layers/channel_mlp.py).

We still lack this functionality in `PointwiseMLP`.

Here is a guide on how to use [equinox.nn.Dropout](https://docs.kidger.site/equinox/api/nn/dropout/).

The respective `channel_mlp_dropout` parameter should also be added to all affected classes that use it.

**Comment by paulgekeler** [comment]
Hi @vbhavh,
I have included this in the new release which unfortunately took me a long time to complete. I will mark this as complete now. Feel free to contribute to other parts of neojax in the future. Would be very much appreciated.
